great_expectations/data_context/store/tuple_store_backend.py

```diff
@@ -559,10 +559,10 @@ def _get_all(self) -> list[Any]:
     def _get_by_s3_object_key(self, s3_client, s3_object_key):
         try:
             s3_response_object = s3_client.get_object(Bucket=self.bucket, Key=s3_object_key)
-        except (s3_client.exceptions.NoSuchKey, s3_client.exceptions.NoSuchBucket):
+        except (s3_client.exceptions.NoSuchKey, s3_client.exceptions.NoSuchBucket) as e:
             raise InvalidKeyError(  # noqa: TRY003
                 f"Unable to retrieve object from TupleS3StoreBackend with the following Key: {s3_object_key!s}"  # noqa: E501
-            )
+            ) from e
```

Comment:
There's a ruff rule for this but we have it disabled due to the number of violations.
We should enable it and noqas for the existing violations.

https://github.com/great-expectations/great_expectations/blob/09816c92c4ecbbe6fcafa96e5cd5d93c057efe4b/pyproject.toml#L321

Reply:
I'm going to merge this. We can do that in a different PR.